lib/simple_map_reduce/server/worker.rb

```diff
@@ -18,7 +18,7 @@ class Worker
       STATES = %i(ready reserved working).freeze
 
       aasm do
-        after_all_transitions :save_state
+        before_all_events :save_state
```

Comment:
This is for avoiding calling `save_state` in the case of directly changing status, like `worker.aasm.current_state = :reserve`